### ISSUE TITLE
"ERROR Type literal has only a call signature -- use" suggesting invalid ts code

### ISSUE THREAD

**Issue by misterdev** [palantir/tslint#4523]
### Bug Report

- __TSLint version__: 5.12.1
- __TypeScript version__: 3.3.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
declare var process: {
	cwd: Function;
	webpackModule: {
		validate: Function;
		WebpackOptionsValidationError: {
			new(errors: string[]): {
				message: string;
			},
		};
	};
	stdout: {
		write: Function;
	};
	exitCode: number;
};
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
      "indent": [
        true,
        "tabs",
        4
      ],
      "ban-types": [
        false
      ],
      "no-console": [true, "log"]
    },
    "rulesDirectory": [],
    "linterOptions": {
      "exclude": [
        "node_modules/**",
        "packages/*/node_modules/**"
      ]
    }
}

```

#### Actual behavior

It gives me
```
 ERROR: packages/migrate/index.ts:20:4 - Type literal has only a call signature — use `new(errors: string[]) => {
                                 message: string;
                         },` instead.
```

That is not valid typescript

```
packages/migrate/index.ts:20:26 - error TS1005: ':' expected.
    new(errors: string[]) => {
```

#### Expected behavior
I think is should suggest to use:
```
new: (errors: string[]) => {
     message: string;
};
```

Please, correct me if I'm wrong.

**Comment by JoshuaKGoldberg** [comment]
Hey @misterdev! This looks like a duplicate of #4291, which was fixed in [5.12.0](https://github.com/palantir/tslint/releases/tag/5.12.0). If you update to that version, the incorrect code suggestion should go away.

Closing this issue as duplicate as I can't duplicate it locally, but please do post again if it still shows up!
